runtime/lua/vim/treesitter/language.lua

```diff
@@ -60,6 +60,10 @@ function M.add(lang, opts)
     filetype = { filetype, { 'string', 'table' }, true },
   })
 
+  if filetype == '' then
+    error(string.format("'%s' is not a valid filetype", filetype))
```

Comment:
could use a validtor function with vim.validate() above

Reply:
Just tried it, and couldn't get it to give a nice enough error message:

```
Passed in:
(string) '.../language.lua:0: filetype: expected filetype or list of filetypes, got '
Expected:
(string) '.../language.lua:0: '' is not a valid filetype'
```

Reply:
> filetype: expected filetype or list of filetypes, got 

This is our standard error format, it's nice because it shows the parameter name clearly as the first item. We could enhance vim.validate() to put quotes around string values, so that it looks like this?

> filetype: expected filetype or list of filetypes, got ""

Reply:
I'm not sure, I think the error message is still worse (standard or not) as we want to explicitly disallow `''`. The standard error message doesn't make that clear enough.

Reply:
I'll merge as is for now, but if the suggested change to `vim.validate` is done (with maybe other changes), then we can update this error message.

One idea would be if a validation function returns the error message to allow more specific error messages, so we could have both depending on input.

```
(string) '.../language.lua:0: filetype: '' is not a valid filetype'
(string) '.../language.lua:0: filetype: expected string or table'
```

Reply:
Sounds good, with the caveat that the message should read:

    (string) '.../language.lua:0: filetype: invalid filetype: ''

It is generally a good idea to put the variable part at the end, else random input can create hard to read messages. It's also more regular and visually predictable for users. This is also the format used for API validation.